tests.py

```diff
@@ -372,6 +373,23 @@ def test_uri_template_matcher_duplicate_variable_error():
         UriTemplateMatcher(u'/foo/{var}/bar/{var}')
 
 
+@mark.parametrize('lval, rval, expected', [
+    (None, [('n1', 'v1'), ('n2', 'v2')], [('n1', 'v1'), ('n2', 'v2')]),
+    ([('n1', 'v1'), ('n2', 'v2')], None, [('n1', 'v1'), ('n2', 'v2')]),
+    ([('n1', 'v1'), ('n2', 'v2')], [('n1', 'v3'), ('n2', 'v4')],
+     [('n1', 'v1'), ('n2', 'v2'), ('n1', 'v3'), ('n2', 'v4')]),
+    (None, None, None),
+])
+def test_uri_template_match_result_update(lval, rval, expected):
+    lval = UriTemplateMatchResult(lval)
```

Comment:
I don't think variable shadowing is a good idea. How about renaming it like`lval_result` instead?